documentation/specs/CodeVisualizer.md

```diff
@@ -0,0 +1,135 @@
+# `CodeVisualizer` Component Specification
+
+## Design
+
+`CodeVisualizer` wraps one or more `CodeBlock`s with a header to allow the user to switch between blocks.
+
+`CodeVisualizer` will be a controlled component. It will require the consumer to provide a selected language and manage its state.
```

Comment:
how do we feel about it being only controlled? only controlled simplifies our implementation but may be a little more friction for the consumer. alternatively we could implement for it allowing both controlled and uncontrolled at the cost of a little more implementation complexity

Reply:
im personally a fan of it only being controlled based on how im understanding the use case

Reply:
> im personally a fan of it only being controlled based on how im understanding the use case

+1